src/injectProps.js

```diff
@@ -2,7 +2,7 @@ function injectProps(target, name, descriptor) {
   const oldFunction = descriptor.value;
 
   descriptor.value = function propsInjectorFunction(...args) {
-    return oldFunction.bind(this)(...args, this.props);
+    return oldFunction.bind(this)(this.props, ...args);
```

Comment:
> In that case I would suggest that the props are passed always as the first argument as otherwise the argument index would vary depending on the number of arguments which the function was invoked with.

you are talking about optional parameters right? This will bump the version to v1.2 but I agree.